libkatana_python_native/src/EntityTypeManager.cpp

```diff
@@ -103,5 +120,42 @@ katana::python::InitEntityTypeManager(py::module_& m) {
             } else {
               return py::cast(EntityType{self, id});
             }
-          });
+          })
+      .def(
+          "get_atomic_subtypes",
+          [](const katana::EntityTypeManager* self, EntityType type) {
+            py::set ret;
+            if (self->HasEntityType(type.type_id)) {
+              auto type_set = self->GetAtomicSubtypes(type.type_id);
+              for (EntityTypeID subtype_id = 0; subtype_id < type_set.size();
+                   ++subtype_id) {
+                if (type_set.test(subtype_id)) {
+                  if (self->GetAtomicTypeName(subtype_id).has_value()) {
+                    ret.add(AtomicEntityType{{self, subtype_id}});
```

Comment:
You don't need the check. The type is an atomic type since it was returned from `GetAtomicSubtypes`.